src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -153,12 +154,12 @@ public DiagnosticBuilderInternal(TextSpan textSpan)
             public ErrorDiagnostic OutputTypeMismatch(object expectedType, object actualType) => new ErrorDiagnostic(
                 TextSpan,
                 "BCP026",
-                $"The output expects a value of type '{expectedType}' but the provided value is of type '{actualType}'.");
+                $"The output expects a value of type {expectedType} but the provided value is of type {actualType}.");
```

Comment:
>{expectedType} [](start = 53, length = 14)

Why remove the single quotes?

Reply:
We should probably discuss, but if we're going to decide to remove all the quotes, we should make all the errors consistent. (There are some remaining that have quotes still.)

---
In reply to: [488332917](https://github.com/Azure/bicep/pull/475#discussion_r488332917) [](ancestors = 488332917)

Reply:
I'm only making this change on types to disambiguate between a type named "MyType" (`MyType`) and a string literal type containing the sting "MyType" (`'MyType'` - including the quotes).

Reply:
Yeah having double single quotes is an issue, but I think we should be consistent on the formatting of error message parameters. With this we end up with some messages not having enclosing quotes and others not. When there aren't enclosing quotes (or some other character) around a word, it looks like it belongs in the sentence structure of the current locale, which it might not when we localize.

Feels like a separate problem that could be addressed separately in all the messages too.

Reply:
That's a good point. Maybe we should use double quotes for quoting things in error messages to avoid confusion with the Bicep character?

Reply:
Yeah was thinking of " as well. Might be ok.